### PR TITLE
Fix diagrams not updating properly

### PR DESCRIPTION
**Pull request by joltcoke**
Hi Chris,

hier mein Versuch das Problem mit den Diagrammen zu fixen. Details hier:

http://knx-user-forum.de/forum/supportforen/cometvisu/828173-diagram-zeitlicher-verlauf-über-x-tage-kaputt

Grüße,
    Florian

### CODE REVIEW

File: src/plugins/diagram/structure_plugin.js [hunk 1]
```diff
@@ -303,20 +303,20 @@ define( ['structure_custom',
 
       // plot diagram initially with empty values
       diagram.empty();
-      data.plot = $.plot(diagram, [], options);
+      var plot = $.plot(diagram, [], options);
```
Comment (ChristianMayer) [inline src/plugins/diagram/structure_plugin.js]: Not keeping a reference (in data) to "plot" anymore

File: src/plugins/diagram/structure_plugin.js [hunk 2]
```diff
@@ -422,10 +423,9 @@ define( ['structure_custom',
               }
 
               // plot
-              var PLOT = data.plot;
-              PLOT.setData(fulldata);
-              PLOT.setupGrid();
-              PLOT.draw();
+              plot.setData(fulldata);
```
Comment (ChristianMayer) [inline src/plugins/diagram/structure_plugin.js]: using "plot" that's not defined anymore?

File: src/plugins/diagram/structure_plugin.js [hunk 3]
```diff
@@ -434,9 +434,9 @@ define( ['structure_custom',
 
       if (data.refresh) {
         // reload regularly
-        window.setTimeout(function( id ) {
-          loadDiagramData( id );
-        }, data.refresh * 1000, id );
+        window.setTimeout(function( id, plot, isInteractive ) {
```
Comment (ChristianMayer) [inline src/plugins/diagram/structure_plugin.js]: Also using plot that's not defined anymore?

File: src/plugins/diagram/structure_plugin.js (hunk 1 above)
Comment (joltcoke) [inline src/plugins/diagram/structure_plugin.js]: Yes, data is shared between the popup and the non-popup version of the diagram. plot is diagram specific and can not be shared, so i needed to factor it out of the data structure.

File: src/plugins/diagram/structure_plugin.js (hunk 2 above)
Comment (joltcoke) [inline src/plugins/diagram/structure_plugin.js]: plot gets now passed as a function argument (see line 358)

File: src/plugins/diagram/structure_plugin.js (hunk 3 above)
Comment (joltcoke) [inline src/plugins/diagram/structure_plugin.js]: Same here.

**Comment by ChristianMayer** [comment]
Ah, I should have looked at the full file... :)

**Pull request merged**